api/v1beta2/authpolicy_types.go

```diff
@@ -57,7 +57,7 @@ type CommonAuthRuleSpec struct {
 	// At least one selected HTTPRoute rule must match to trigger the auth rule.
 	// If no route selectors are specified, the auth rule will be evaluated at all requests to the protected routes.
 	// +optional
-	// +kubebuilder:validation:MaxItems=15
+	// +kubebuilder:validation:MaxItems=8
```

Comment:
Sure, can add a test. Though, I'm expecting this value to decrease again when the overrides field is added in https://github.com/Kuadrant/kuadrant-operator/issues/464

Reply:
Cool, and thinking out loud. If we need to increase that number in the future, is the workaround to create a webhook to do the validation and drop the use of the CEL?

Reply:
Added integration test for this. 

I don't think just simply switching to using validation webhook would work, as I believe this more so relates to using a slice of `RouteSelectors` that the fields itself also have CEL validation for and how frequently we have this slice of `RouteSelectors` in the various structs used within the CRD. This will fail the CEL cost estimation when applying the CRD even if switch to using a validating webhook 🤔 

https://github.com/Kuadrant/kuadrant-operator/blob/4f6a138d4b670381f6ef83409f5bffc2824c19ff/api/v1beta2/route_selectors.go#L13-L24

Reply:
It's because we're importing Gateway API types and, with those, their CEL validation as well.

I guess our best options to avoid this hard limit are:
1. wait for https://github.com/kubernetes-sigs/gateway-api/pull/2593 to be implemented and then simplify the route selectors
2. redefine the HTTPRouteMatch types ourselves without the validation – we could do this because the validation that matters is the one enforced on the HTTPRoute, not on the policy; on the policy, the only consequence of an "invalid" route selector is that the selector will not select any route rule.